lib/connection_string.js

```diff
@@ -706,6 +713,22 @@ function parseConnectionString(uri, options, callback) {
     return callback(new MongoParseError('No hostname or hostnames provided in connection string'));
   }
 
+  const directConnection = !!parsedOptions.directConnection;
+  if (directConnection && hosts.length !== 1) {
+    // If the option is set to true, the driver MUST validate that there is exactly one host given
+    // in the host list in the URI, and fail client creation otherwise.
+    return callback(new MongoParseError('directConnection option requires exactly one host'));
+  }
+
+  // NOTE: this behavior will go away in v4.0, we will always auto discover there
```

Comment:
this _is_ v4.0, so it seems like something should change here?

Reply:
[This block is in the 3.6 commit](https://github.com/mongodb/node-mongodb-native/commit/c5d60fc4619227697ef2102437fe5c8b111909d2#diff-8a39aaa3a2d1338e1a7455cd10b47ae2R683-R698). I think you might have added it. I've tried to remove it but spec fails. I've tried to learn more about [auto-discovery but all i've found in the spec is line](https://github.com/mongodb/specifications/search?q=auto-discovery&unscoped_q=auto-discovery):

> Why is auto-discovery the preferred default?
> Auto-discovery is most resilient and is therefore preferred.

I'm assuming that there's a breaking change we can make in 4.0 that's not in 3.6 but I can't find what that change is or how to execute it.

Reply:
answered on slack